peertube: startup script: exec into nodejs

ppom0 opened the pull request:
So, there is no big deal in this PR, just a little bit of cleaning for peertube's startup script.
Thanks for this NixOS module which saved me lots of time! 

1. Removed the `#!/bin/sh` shebang at the beginning, because `systemd.services.<name>.script` already adds a `#!/nix/store/.../bin/bash` shebang.

Previously:
```bash
#!/nix/store/516kai7nl5dxr792c0nzq0jp8m4zvxpi-bash-5.2p32/bin/bash
set -e
#!/bin/sh
umask 077
...
```

2. Exec into nodejs, so that the startup script is no longer running but replaces itself by nodejs.
This way, only one processus is running inside peertube.service.

ping maintainers: @immae @Izorkin @stevenroose

Atemu reviewed (approved):
This looks simple enough

Atemu commented:
Ah one minor nit: The convention for nixos modules is to prefix `nixos/` to the commit message; i.e. `nixos/peertube: ...`

Izorkin commented:
The `Peertube` test is not working at the moment.

Commit:
nixos/peertube: startup script: exec into nodejs

1. Removed the #!/bin/sh shebang at the beginning, because
systemd.services.<name>.script already adds a #!/nix/store/.../bin/bash
shebang.

Previously:
	#!/nix/store/516kai7nl5dxr792c0nzq0jp8m4zvxpi-bash-5.2p32/bin/bash
	set -e
	#!/bin/sh
	umask 077
	...

2. Exec into nodejs, so that the startup script is no longer running but
replaces itself by nodejs.
This way, only one processus is running inside peertube.service.

Atemu commented:
@ppom0 could you rename the commit? Then we're good to merge.

ppom0 commented:
Done, thanks!
